src/phoenix/server/api/types/TraceSession.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+import strawberry
+from sqlalchemy import desc, select
+from strawberry import UNSET, Info
+from strawberry.relay import Connection, Node, NodeID
+
+from phoenix.db import models
+from phoenix.server.api.context import Context
+from phoenix.server.api.types.pagination import ConnectionArgs, CursorString, connection_from_list
+from phoenix.server.api.types.Trace import Trace, to_gql_trace
+
+
+@strawberry.type
+class TraceSession(Node):
+    id_attr: NodeID[int]
+    session_id: str
+
+    @strawberry.field
+    async def traces(
+        self,
+        info: Info[Context, None],
+        first: Optional[int] = 50,
+        last: Optional[int] = UNSET,
+        after: Optional[CursorString] = UNSET,
+        before: Optional[CursorString] = UNSET,
+    ) -> Connection[Trace]:
+        args = ConnectionArgs(
+            first=first,
+            after=after if isinstance(after, CursorString) else None,
+            last=last,
+            before=before if isinstance(before, CursorString) else None,
+        )
+        stmt = (
+            select(models.Trace)
+            .filter_by(trace_session_id=self.id_attr)
+            .order_by(desc(models.Trace.id))
+            .limit(first)
+        )
```

Comment:
Did we want to do offset-based pagination here? Looks like we're taking the same spans no matter what cursor is passed in for `after`.

Reply:
Going to punt on this. I dug in a little and found it non-trivial to understand everything for the time being. And since i just copied this from the `Trace.spans` resolver, they would share the same issue, and as such should probably be addressed in the same way.

Reply:
Can also just return a list rather than a connection.

Reply:
returning a list would change the query pattern, i.e. removing `edges` etc., so punting on that also